src/components/tsx components/InputPins.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { FormEvent } from "react";
+import styled from "styled-components";
+
+interface InputPinsProps {
+  scoreArray: number[];
+  setScoreArray: React.Dispatch<React.SetStateAction<number[]>>;
+}
+
+export default function InputPins({
+  scoreArray,
+  setScoreArray,
+}: InputPinsProps) {
+  const [inputSubmit, setInputSubmit] = React.useState<string>("");
+
+  function inputScore(event: FormEvent) {
+    event.preventDefault();
+    const pinos: number = parseInt(inputSubmit);
+    const isNumber: boolean = /[0-9]/.test(inputSubmit);
+
+    if (isNumber) {
+      if (scoreArray.length % 2 !== 0) {
+        let reamaningPins = [...scoreArray];
+        let lastNumber = reamaningPins.pop();
+        if (pinos > 10 - (lastNumber ? lastNumber : 0)) {
+          alert("Valor invalido");
+          return;
+        }
+      }
+
+      if (scoreArray.length % 2 === 0 && pinos === 10) {
+        let currentArray = [...scoreArray, pinos, 0];
+        setScoreArray(currentArray);
```

Comment:
We could use the fn signature here i.e. `setScoreArray(prevPlays => [...prevPlays, pinos, 0]);`